cmd/promtool/backfill_test.go

```diff
@@ -260,6 +258,46 @@ http_requests_total{code="400"} 2 1565133715.989
 		{
 			ToParse: `# HELP http_requests_total The total number of HTTP requests.
 # TYPE http_requests_total counter
+http_requests_total{code="200"} 1021 1624463088.000
+http_requests_total{code="200"} 1 1627055153.000
+http_requests_total{code="400"} 2 1627056153.000
```

Comment:
While this is testing the max block duration, we are missing some edge cases. Here are my suggestions for the test cases:

(Reference: `ExponentialBlockRanges(2, 10, 3) -> [2 6 18 54 162 486 1458 4374 13122 39366]`)

1. 3 samples of same series such that 2 of them fall in the same block and 1 in another block. With max block duration same as right now (2048h), is 1458h.

(t, t+1400h, t+1500h)
```
http_requests_total{code="200"} 1 1624463088.000
http_requests_total{code="200"} 2 1629503088.000
http_requests_total{code="200"} 3 1629863088.000
```

2. Same as above, but with max block duration much more than `> 39366h`. So lets keep it `200000h` (basically >3x of largest block).

(t, t+39300h, t+40000h)
```
http_requests_total{code="200"} 1 1624463088.000
http_requests_total{code="200"} 2 1765943088.000
http_requests_total{code="200"} 3 1768463088.000
```

Expected number of blocks in both cases is 2.